pkg/inject/envoy.go

```diff
@@ -187,15 +86,7 @@ func (m *envoyMutator) mutate(pod *corev1.Pod) error {
 		return err
 	}
 	variables := m.buildTemplateVariables(pod)
-	envoySidecar, err := renderTemplate("envoy", envoyContainerTemplate, variables)
-	if err != nil {
-		return err
-	}
-	container := corev1.Container{}
-	err = json.Unmarshal([]byte(envoySidecar), &container)
-	if err != nil {
-		return err
-	}
+	container := buildEnvoySidecar(variables)
```

Comment:
Like the idea. However, if we want to move away from templates then I think we should change how we currently inject envoys in to VirtualGateways as well so that we have uniformity on how we inject envoys - https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/pkg/inject/virtualgateway_envoy.go#L14

Reply:
Also, we had a  brief discussion around this in the past but might be worth adding datadog support for VirtualGateways as well now that we are making a change around this area. Right now we only have support for XrayTracing in VirtualGateways.

Reply:
I'd break those into separate PRs. Virtual gateway is similar but not similar (we "optionally" Envoy and it's config). For the DataDog tracers in Virtual gateways, it should definitely be a separate PR as users have control over virtual gateway envoy spec until virtual nodes